docs/threading-model.md

```diff
@@ -0,0 +1,21 @@
+# Threading model
+
+## Thread pool size
+The thread pool size initialization is written in the function `init_once()`.\
```

Comment:
You shall describe how dcurl makes use of thread pool

Reply:
I added some brief description of `dcurl` and the thread pool.